server/controllers/shoppingList.js

```diff
@@ -205,24 +158,24 @@ const getListData = (req, resp) => {
         { purchaserIds: req.user._id }
       ]
     },
-    (err, documents) => {
+    (err, doc) => {
       if (err) {
-        return resp.status(404).send({ message: err.message });
+        return resp.status(400).send({
+          message:
+            'An error occurred while fetching the lists data. Please try again.'
+        });
       }
 
-      if (documents && documents.length > 0) {
-        const { cohortId, _id, isArchived } = documents[0];
-
-        if (isArchived) {
-          return resp.status(200).json({ cohortId, _id, isArchived });
-        }
+      if (doc) {
+        const list = _head(doc);
```

Comment:
We look exactly for one list, so we can use findOne and use doc without lodash _head an or checking docs.length